.ci/dockerfiles/Dockerfile.xpu

```
@@ -0,0 +1,134 @@
+RUN git clone ${UCX_REPO_URL} -b ${UCX_BRANCH} ${UCX_SRC_DIR} && \
+    cd ${UCX_SRC_DIR} && \
+    ./autogen.sh && \
+    ./configure --prefix=${UCX_INSTALL_DIR} --with-mlx5=no --with-ze=yes --enable-examples --enable-mt && \
+    make -j$(nproc) && \
+    make -j install-strip && \
+    ldconfig
+
+WORKDIR ${NIXL_SRC_DIR}
+COPY . .
+
+RUN set -eux; \
```

Hadolint (2.14.0)

[warning] Pin versions in pip. Instead of `pip install <package>` use `pip install <package>==<version>` or `pip install --requirement <requirements file>`

(DL3013)